New Lucas just Dropped

HisashiASM opened the pull request:
- Added Offense Up
    - Windbox on OU Charge on frame 14
    - Offense Up charges for 2 seconds - Retains charge when hit out or special button is released
    - Offense Up is lost when whiffing Smashes, Death, Reset or OU Burst
    - Added OU Burst and OU Burst Freeze
    - Added new Smashes for Offense Up
         - USmash is OU Explosion, a HUGE explosion that deals 24% damage to a player, yet may be harder to hit floored opponents
         - Side Smash has an increased tipper and deals more damage
         - Down smash has a larger particle explosion, making it easier to get DACDS and is a little stronger
     - SFX and VFX Changes to a load of moves to fit the new OU stuff
     - DTaunt then press and hold shield on Training to quickly enable OU
   
- Added MagPulling
    - After jumping once and pressing sspecial, press magnet to gain a small speed boost going in the direction you magpulled, it is possible to b reverse if timed correctly

- Added Tiltable Side Smash
- UTaunt goes into Slip if held and now has a new animation
- SSpecial is now PK Freeze, but acts as PK Fire with more range and a smaller KB and damage
    - Now uses different voicelines
    - Shortened PK Freeze timings for output of the particle while in air
- DAir readjusted to fit new design
- Shortened DSmash hits from 3 to 1
- Shortened USmash (uncharged) times to end hitboxes on f43
- Fixed audio bug on Jab 1-2-3
- Back Air is less strong and has some of the hitbox IDs readjusted, also there is an angle change
- Up Air now comes out differently animation wise, and has a new launch angle and kb
- Forward air now has a bodybox

Needs HDR-Development/hdr-private#168 and HDR-Development/hdr-private#177

HisashiASM committed:
Added a lot - New Lucas just Dropped

+ Added Offence Up
+ Added new Smash's for Offence Up
+ Added Tiltable Forward Smash
+ Added OU Burst and OU Burst Freeze
+ UTaunt causes Slip if held
* DAir readjusted to fit new design
* Shortened DSmash hits from 3 to 1
* Shortened USmash (uncharged) times to end hitboxes on f43

HisashiASM committed:
Merge branch 'dev' into Lucas-Rework

HisashiASM committed:
Fixed VarModule Calls

Thank you WuBoy.

HisashiASM committed:
Fix Const

Am bozo, repairs smash tiling.

HisashiASM committed:
Bozo forgor angles

Yeah, I am on that 12:00AM brain

HisashiASM committed:
Tons of changes

* Changed sound time to be equal to whiff flag
* OU Charge time reduced from 3 frames to 2
* Fixed a bug causing Lucas to uncharge OU on FSMASH
* Increased the dair lower hitbox size
* Fixed a bug preventing the OU effect from showing when hit out of charge
* Adjusted joint rotator logic

HisashiASM committed:
Whoops prints and damage fix

Sorry lol

HisashiASM committed:
Adjusted with the assist of MadVillian pt.1

All guchii thanks for the help!

HisashiASM committed:
Effects Fixes

Riah321 commented:
New mechanics feel really good. Offense up seems fair in it's current state.
Down air feels really jank but probably skill issue. Super hype for this rework :)
Down taunt transitioning to trip is a little wonky in the animation.

HisashiASM committed:
SFX Pushes and MagPulling

Here we go, maybe the last changes?

HisashiASM committed:
Updated handing for offence up, more accuracy by checking in opff

As the title says. Also added some info to my updated version of techy's method of rotating bones if you care.

jobrien97 commented:
Please edit AttackLw4's `expression` ACMD script to account for the new single-hit functionality. Controller still rumbles 3 times.

HisashiASM committed:
Special N Default Status Script - Do Not Install

Includes some other stuff for future revisions.

HisashiASM committed:
Pre-Final with more to hdr-assets

Some things will feel off because you will not have access to the new romfs files. Expect bugs.

HisashiASM committed:
Final Effect Changes!

HisashiASM committed:
Little Cleanup on Bair and Better USmash OU particletime.

jobrien97 commented on fighters/lucas/src/acmd/specials.rs:
```diff
@@ -238,59 +230,182 @@ unsafe fn lucas_special_lw_end_effect(fighter: &mut L2CAgentBase) {
+        if is_excute(fighter) {
+            if i%2==0 {
+                EFFECT_OFF_KIND(fighter, Hash40::new("lucas_pkfr_hold"), false, false);
+                EFFECT_FLW_POS(fighter, Hash40::new("lucas_pkfr_hold"), Hash40::new("top"), 0, sv_math::rand(hash40("fighter"), 4) as i32 + 12, sv_math::rand(hash40("fighter"), 4) as i32 - 2, 0, 0, 0, 0.5, true);
```
You can use the `emit` bone to make an effect spawn randomly on a character's body

jobrien97 commented on fighters/lucas/src/opff.rs:
```diff
@@ -212,15 +212,170 @@ unsafe fn pk_fire_ff(boma: &mut BattleObjectModuleAccessor, stick_y: f32) {
+// Offense Up charge Handler
+pub unsafe fn offense_charge(fighter: &mut smash::lua2cpp::L2CFighterCommon, boma: &mut BattleObjectModuleAccessor, situation_kind: i32)  {
+    if(VarModule::is_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_ACTIVE)) {
+        if fighter.is_status(*FIGHTER_STATUS_KIND_SPECIAL_N) {
+            StatusModule::change_status_request_from_script(fighter.module_accessor, *FIGHTER_LUCAS_STATUS_KIND_SPECIAL_N_FIRE, true);
+        }
```
This should be moved to SPECIAL_N's pre status, using set_status_kind_interrupt if the flag is on. Else, call original.
Doing this in opff causes the change_status_request_from_script to run on frame 2 of SPECIAL_N, as status_kind isn't updated until then

jobrien97 commented on fighters/lucas/src/status.rs:
```diff
@@ -2,12 +2,259 @@ use super::*;
+        }
+    }
+}
+
```
pls comment `// FIGHTER_LUCAS_STATUS_KIND_SPECIAL_N_HOLD //` here, to keep consistent with our status script organization

HisashiASM committed:
Code Refactor - Sound on S Special Collision to be Ice

Renamed methods to match code base, changed fire logic, added the ice sfx to s special pillar to sound like an ice attack.

HisashiASM committed:
Bair nerf body box, fix interpolation

FINAL FIX! PLEASE!

jobrien97 reviewed (approved):
lgtm :D
